src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs

```diff
@@ -65,11 +79,22 @@ public async Task WarmupInvoke(HttpContext httpContext)
 
             ReadRuntimeAssemblyFiles();
 
-            await WarmUp(httpContext.Request);
+            await WorkerWarmup();
```

Comment:
Still trying to understand this change after looking at the sequential diagram from the design linked. One question: Why can't the language worker do this within the worker init request handling itself?

Who makes a call to the host warmup middleware and when does that happen?

Reply:
DWAS and its equivalent in Linux makes the warmup call every few minutes while still in placeholder mode. This is not a one time call and should happen every few minutes in placeholder mode to avoid OS paging as much as possible during specialization.